v2/component/launcher.go

```diff
@@ -170,8 +173,33 @@ const defaultPipelineRoot = "minio://mlpipeline/v2/artifacts"
 // NewLauncher creates a new launcher object using the JSON-encoded runtimeInfo
 // and specified options.
 func NewLauncher(runtimeInfo string, options *LauncherOptions) (*Launcher, error) {
+	restConfig, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, fmt.Errorf("Failed to initialize kubernetes client: %w", err)
+	}
+	k8sClient, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to initialize kubernetes client set: %w", err)
+	}
+	namespace := os.Getenv("KFP_NAMESPACE")
+	if namespace == "" {
+		return nil, fmt.Errorf("Env variable 'KFP_NAMESPACE' is empty")
+	}
+
 	if len(options.PipelineRoot) == 0 {
 		options.PipelineRoot = defaultPipelineRoot
+		config, err := getLauncherConfig(k8sClient, namespace)
+		if err != nil {
+			return nil, err
+		}
+		// LauncherConfig is optional, so it can be nil when err == nil.
+		if config != nil && config.Data != nil {
+			defaultRootFromConfig := config.Data["defaultPipelineRoot"]
```

Comment:
Is it possible that user defines a configmap but doesn't define the "defaultPipelineRoot" in configmap? Should we throw a error here?

Reply:
I'd suggest we treat all configuration with maximum tolerance, e.g.:

* everything should have a default value
* everything is optional

Therefore, I don't throw an error here.

I did add a `glog.Infof` afterwards to make it obvious what we defaults to in the end. (our logging doesn't show info logs by default, but we can fix that separately)

What do you think about these choices?

Reply:
sgtm